pkg/predicate/predicate_test.go

```diff
@@ -516,4 +516,39 @@ var _ = Describe("Predicate", func() {
 			})
 		})
 	})
+
+	Describe("When checking a LabelSelectorPredicate", func() {
+		instance, err := predicate.LabelSelectorPredicate(*(&metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}))
+		if err != nil {
+			Fail("Improper Label Selector passed during predicate instantiation.")
+		}
+
+		Context("When the Selector does not match the event labels", func() {
+			It("should return false", func() {
+				failMatch := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"bar": "foo"},
```

Comment:
Nit: Just using `&corev1.Pod{}` would be enough, no need to give ti a different set of labels